File: find-english-only-calculators.py
import re
import json

# Read the calculators.ts file
with open(r'c:\Users\Hammad Razi\Desktop\the-smart-calculator\meta\calculators.ts', 'r', encoding='utf-8') as f:
    content = f.read()

# Find all calculator entries
# Pattern to match calculator IDs and their language blocks
calculator_pattern = r"'([^']+)':\s*\{([^}]+(?:\{[^}]*\}[^}]*)*)\},"

matches = re.finditer(calculator_pattern, content, re.DOTALL)

english_only_calculators = []
multi_language_calculators = []

for match in matches:
    calc_id = match.group(1)
    calc_content = match.group(2)
    
    # Check which languages are present
    has_en = 'en:' in calc_content
    has_pl = 'pl:' in calc_content
    has_br = 'br:' in calc_content
    has_de = 'de:' in calc_content
    has_es = 'es:' in calc_content
    
    # Extract slug from en block
    slug_match = re.search(r'en:\s*\{[^}]*slug:\s*["\']([^"\']+)["\']', calc_content, re.DOTALL)
    slug = slug_match.group(1) if slug_match else None
    
    if has_en and not (has_pl or has_br or has_de or has_es):
        english_only_calculators.append({
            'id': calc_id,
            'slug': slug
        })
    else:
        multi_language_calculators.append({
            'id': calc_id,
            'slug': slug
        })

# Print results
print("=" * 80)
print(f"ENGLISH-ONLY CALCULATORS: {len(english_only_calculators)}")
print("=" * 80)
for calc in english_only_calculators:
    print(f"  '{calc['slug']}',  // {calc['id']}")

print("\n" + "=" * 80)
print(f"MULTI-LANGUAGE CALCULATORS: {len(multi_language_calculators)}")
print("=" * 80)

# Save to JSON for further processing
output = {
    'english_only': english_only_calculators,
    'multi_language': multi_language_calculators
}

with open(r'c:\Users\Hammad Razi\Desktop\the-smart-calculator\calculator-analysis.json', 'w', encoding='utf-8') as f:
    json.dump(output, f, indent=2)

print(f"\n✅ Analysis saved to calculator-analysis.json")
print(f"   - English-only: {len(english_only_calculators)}")
print(f"   - Multi-language: {len(multi_language_calculators)}")

File: find-english-only-calculators.js
const fs = require('fs');

// Read the calculators.ts file
const content = fs.readFileSync('meta/calculators.ts', 'utf-8');

// Split by calculator entries
const lines = content.split('\n');

let currentCalc = null;
let currentBlock = '';
let inCalcBlock = false;
let braceCount = 0;

const calculators = {};

for (let i = 0; i < lines.length; i++) {
  const line = lines[i];
  
  // Match calculator ID
  const calcMatch = line.match(/^\s*'([^']+)':\s*\{/);
  if (calcMatch) {
    currentCalc = calcMatch[1];
    calculators[currentCalc] = { languages: [], slug: null };
    inCalcBlock = true;
    braceCount = 1;
    currentBlock = line;
    continue;
  }
  
  if (inCalcBlock) {
    currentBlock += '\n' + line;
    
    // Count braces
    braceCount += (line.match(/\{/g) || []).length;
    braceCount -= (line.match(/\}/g) || []).length;
    
    // Check for language blocks
    if (line.match(/^\s*(en|pl|br|de|es):\s*\{/)) {
      const lang = line.match(/^\s*(en|pl|br|de|es):/)[1];
      calculators[currentCalc].languages.push(lang);
    }
    
    // Extract slug from en block
    if (line.includes('slug:') && line.includes('"')) {
      const slugMatch = line.match(/slug:\s*"([^"]+)"/);
      if (slugMatch) {
        calculators[currentCalc].slug = slugMatch[1];
      }
    }
    
    // End of calculator block
    if (braceCount === 0) {
      inCalcBlock = false;
      currentCalc = null;
      currentBlock = '';
    }
  }
}

// Separate English-only and multi-language calculators
const englishOnly = [];
const multiLanguage = [];

for (const [id, data] of Object.entries(calculators)) {
  if (data.languages.length === 1 && data.languages[0] === 'en') {
    englishOnly.push({ id, slug: data.slug });
  } else if (data.languages.length > 0) {
    multiLanguage.push({ id, slug: data.slug, languages: data.languages });
  }
}

// Print results
console.log('='.repeat(80));
console.log(`ENGLISH-ONLY CALCULATORS: ${englishOnly.length}`);
console.log('='.repeat(80));
englishOnly.forEach(calc => {
  console.log(`  '${calc.slug}',  // ${calc.id}`);
});

console.log('\n' + '='.repeat(80));
console.log(`MULTI-LANGUAGE CALCULATORS: ${multiLanguage.length}`);
console.log('='.repeat(80));

// Save to JSON
const output = {
  english_only: englishOnly,
  multi_language: multiLanguage
};

fs.writeFileSync('calculator-analysis.json', JSON.stringify(output, null, 2));

console.log(`\n✅ Analysis saved to calculator-analysis.json`);
console.log(`   - English-only: ${englishOnly.length}`);
console.log(`   - Multi-language: ${multiLanguage.length}`);
